app/src/ui/history/commit-summary.tsx

```diff
@@ -316,17 +339,145 @@ export class CommitSummary extends React.Component<
   }
 
   private getShaRef = (useShortSha?: boolean) => {
-    const { commits } = this.props
-    const oldest = useShortSha ? commits[0].shortSha : commits[0].sha
+    const { selectedCommits } = this.props
+    return useShortSha ? selectedCommits[0].shortSha : selectedCommits[0].sha
+  }
+
+  private onHighlightShasInDiff = () => {
+    this.props.onHighlightShas(this.props.shasInDiff)
+  }
+
+  private onHighlightShasNotInDiff = () => {
+    const { onHighlightShas, selectedCommits, shasInDiff } = this.props
+    onHighlightShas(
+      selectedCommits.filter(c => !shasInDiff.includes(c.sha)).map(c => c.sha)
+    )
+  }
+
+  private onRemoveHighlightOfShas = () => {
+    this.props.onHighlightShas([])
+  }
+
+  private showUnreachableCommits() {
+    // TODO: open to dialog with commits list of unreachable commits
+  }
+
+  private renderCommitsNotReachable = () => {
+    const { selectedCommits, shasInDiff } = this.props
+    if (selectedCommits.length === 1) {
+      return
+    }
+
+    const excludedCommitsCount = getCountCommitsNotInDiff(
+      selectedCommits,
+      shasInDiff
+    )
+
+    if (excludedCommitsCount === 0) {
+      return
+    }
+
+    return (
+      <div
+        className="commit-unreachable-info"
+        onMouseOver={this.onHighlightShasNotInDiff}
+        onMouseOut={this.onRemoveHighlightOfShas}
+      >
+        <Octicon symbol={OcticonSymbol.info} />
+        <LinkButton onClick={this.showUnreachableCommits}>
+          {excludedCommitsCount} unreachable commits
```

Comment:
We should handle plural vs singular here right?
![image](https://user-images.githubusercontent.com/1083228/176642839-57ec82bb-cbb4-4046-ae5c-1667a5beedae.png)

Reply:
Yes. 🤦